lib/PluginHandler.js

```diff
@@ -115,39 +113,33 @@ class PluginHandler {
      *
      * @param {string} name name of the plugin
      * @param {Record<string, any>} parentConfig io-package of the parent module that uses the plugins (adapter/controller)
-     * @param {(error?: string) => void} [callback] callback function which is called after initialization is done for all plugins
      */
-    initPlugin(name, parentConfig, callback) {
+    async initPlugin(name, parentConfig) {
         const instance = this.plugins[name].instance;
         if (!instance) {
-            typeof callback === 'function' && callback(new Error('Please instanciate plugin first!'));
-            return;
+            throw new Error('Please instantiate plugin first!');
         }
-        instance.initPlugin(this.plugins[name].config, parentConfig, (err, initSuccessful) => {
-            if (err || !initSuccessful) {
-                this.log.debug('Plugin ' + name + ' destroyed because not initialized correctly');
 
-                instance.destroy();
-                delete this.plugins[name].instance;
-            }
-            typeof callback === 'function' && callback();
-        })
+        try {
+             await instance.initPlugin(this.plugins[name].config, parentConfig)
+        } catch (e) {
+            this.log.debug(`Plugin ${name} destroyed because not initialized correctly`);
+
+            instance.destroy();
+            delete this.plugins[name].instance;
+        }
     }
 
     /**
      * Initialize all Plugins that are registered
      *
      * @param {Record<string, any>} parentConfig io-package of the parent module that uses the plugins (adapter/controller)
-     * @param {(error?: string) => void} callback callback function which is called after initialization is done for all plugins
      */
-    initPlugins(parentConfig, callback) {
-        let callbackCnt = 0;
-        Object.keys(this.plugins).forEach(plugin => {
-            if (!this.plugins[plugin].instance) return;
-            callbackCnt++;
-            this.initPlugin(plugin, parentConfig, () => !--callbackCnt && typeof callback === 'function' && callback());
-        });
-        callbackCnt === 0 && typeof callback === 'function' && callback();
+    async initPlugins(parentConfig) {
+        for (const [pluginName, plugin] of Object.entries(this.plugins)) {
+            if (!plugin.instance) return;
+            await this.initPlugin(pluginName, parentConfig);
```

Comment:
I think doing them sequencially is ok ... could also be done in oparallel as before

Reply:
We can also go for promise.all but I guess it won't have an significant impact. Okay rn we only have sentry but also in general. If you want I can change but I would personally keep it.